rust_src/src/editfns.rs

```diff
@@ -49,3 +50,49 @@ pub fn bobp() -> LispObject {
     let buffer_ref = ThreadState::current_buffer();
     LispObject::from_bool(buffer_ref.pt == buffer_ref.begv)
 }
+
+// Return the start or end position of the region.
+// BEGINNINGP means return the start.
+// If there is no region active, signal an error.
+fn region_limit(beginningp: bool) -> LispObject {
+    let current_buf = ThreadState::current_buffer();
+    if LispObject::from_raw(unsafe { globals.f_Vtransient_mark_mode }).is_not_nil() &&
+        LispObject::from_raw(unsafe { globals.f_Vmark_even_if_inactive }).is_nil() &&
+        current_buf.mark_active().is_nil()
+    {
+        xsignal!(Qmark_inactive);
+    }
+
+    let m = marker_position(current_buf.mark());
+    if m.is_nil() {
+        error!("The mark is not set now, so there is no region");
+    }
+
+    let num = m.as_fixnum_or_error();
+    // Clip to the current narrowing (bug#11770)
+    LispObject::from_fixnum(if ((current_buf.pt as EmacsInt) < num) == beginningp {
```

Comment:
I would personally find this more readable if we pulled the if statement out of the from_fixnum call